packages/@aws-cdk/assets/lib/fs/fingerprint.ts

```diff
@@ -38,49 +42,64 @@ export interface FingerprintOptions {
  * @param options Fingerprinting options
  */
 export function fingerprint(fileOrDirectory: string, options: FingerprintOptions = { }) {
-  const follow = options.follow !== undefined ? options.follow : FollowMode.External;
-  const hash = crypto.createHash('md5');
-  addToHash(fileOrDirectory);
+  const hash = crypto.createHash('sha256');
+  _hashField(hash, 'options.extra', options.extra || '');
+  const follow = options.follow || FollowMode.External;
+  _hashField(hash, 'options.follow', follow);
 
-  hash.update(`==follow==${follow}==\n\n`);
-
-  if (options.extra) {
-    hash.update(`==extra==${options.extra}==\n\n`);
-  }
-
-  for (const ex of options.exclude || []) {
-    hash.update(`==exclude==${ex}==\n\n`);
-  }
+  const rootDirectory = fs.statSync(fileOrDirectory).isDirectory()
+    ? fileOrDirectory
+    : path.dirname(fileOrDirectory);
```

Comment:
missing tests for all the new options

Reply:
As far as I know, I have not introduced any new options 😅 but yes, some of that surface is untested. I'll se if I can get around to fixing that.

Reply:
The ability to exclude files has been added (previously these options were ignored)